src/uu/split/src/split.rs

```diff
@@ -783,6 +807,29 @@ impl Settings {
                 }
             }
         }
+
+        // Make sure that separator is only one UTF8 character (if specified)
+        // defaults to '\n' - newline character
+        // If the same separator (the same value) was used multiple times - `split` should NOT fail
+        // If the separator was used multiple times but with different values (not all values are the same) - `split` should fail
+        let separator = match matches.get_many::<String>(OPT_SEPARATOR) {
+            Some(sep_values) => match sep_values.map(|s| s.to_string()).reduce(|cur, nxt| {
```

Comment:
A `fold` might be more appropriate here, to avoid the `Option` in the return type.

You could also write the check like this instead of the reduce:
```rust
let first = sep_values.next();
match first {
    Some(first) => {
        sep_values.all(|s| s == first);
        // match on the value here
    }
    None => b'\n',
}
```
I'll leave it up to you what you think looks best in the end.

Reply:
just pushed a new version - still using reduce, but since we can assume that Clap's get_many() call should return non-empty vector inside Some(ValuesRef<'_,String>) - it is safe to just unwrap and skip several match blocks one within another - just have one.